### PR TITLE
First crack at threading `custom_layers` through.

### PR DESCRIPTION
**Pull request by stephenbalaban**
Keras currently uses globals() to access layers. (This doesn't work
across modules.) Hacky solution was to pass a dict mapping name -> class.
I called this dict `custom_layers`. I have a few questions before we merge this in:

1. What would be a better name for this kwarg? `user_layers`? `layers`?
2. Is there a better way of doing this that I'm not seeing?
3. Are there any other places I missed that where should add the `custom_layers` argument?

Currently serialziation with custom layers works fine. The problem that this fixes is trying to call `model_from_*` with a configuration file that has a custom layer name that isn't inside of Keras' `globals()`. This solution allows us to serialize / deserialize models with custom layers. For example:

```py
from keras.models import model_from_yaml
import keras.layers.core

class FooLayer(keras.layers.core.Layer):
    pass

my_layers = {
    'FooLayer': FooLayer
}

with open('foo-network.yaml', 'r') as cfg:
    model = model_from_yaml(cfg.read(), custom_layers=my_layers)
```

### CODE REVIEW

File: keras/utils/generic_utils.py [hunk 1]
```diff
@@ -5,9 +5,9 @@
 import six
 
 
-def get_from_module(identifier, module_params, module_name, instantiate=False, kwargs=None):
+def get_from_module(identifier, module_params, module_name, instantiate=False, kwargs=None, custom_layers={}):
```
Comment (fchollet) [inline keras/utils/generic_utils.py]: You should not be modifying `get_from_module`. Instead you should be merging the `custom_layers` dict with the layer dict from `globals`.

**Comment by fchollet** [comment]
> What would be a better name for this kwarg? user_layers? layers?

`custom_layers` is fine.

**Comment by stephenbalaban** [comment]
What's the advantage of injecting this into `globals()` as opposed to passing in the dictionary explicitly?

**Comment by fchollet** [comment]
`get_from_module` is a generic utility that has nothing to do with model instantiation and should be kept separate. It's used throughout the codebase. That's why it's part of `generic_utils`, not `layer_utils`.

**Comment by stephenbalaban** [comment]
Ah, that makes perfect sense. I misinterpreted the purpose of `get_from_module`. I'll make those changes.
